### ISSUE TITLE
Writes empty files when using sink

### ISSUE THREAD

**Issue by shadowhand** [Kevinrob/guzzle-cache-middleware#82]
Whenever I use this middleware when combined with the [`sink`](http://guzzle.readthedocs.io/en/latest/request-options.html#sink) option empty files are written:

```php
$client->get($uri, ['sink' => "$dir/$uri"]);
```

The file `$dir/$uri` is created but has zero-length content.

**Comment by Kevinrob** [comment]
Are you sure that this is this middleware that cause the issue?  
The `sink` option is handled before this middleware.  

Can you share/post your code?

**Comment by shadowhand** [comment]
Yes, I've confirmed that `sink` works correctly when this middleware is disabled.

Guzzle [config is here](https://github.com/RoundingWellOS/redox-php/blob/feature/json-schema/config/injection/guzzle.php#L22)

Here is our [calling code](https://github.com/RoundingWellOS/redox-php/blob/feature/json-schema/bin/download#L35-L36), which is currently not using `sink` due to this issue.

**Comment by samwilson** [comment]
We're also seeing this issue in [this project](https://github.com/wsexport/tool/blob/05bd30ea20e011b2ebb32e1a107268ed2ea2b9b1/src/BookProvider.php#L219). The workaround of not using `sink` (and instead writing the response body out to a file after the Guzzle request) seems to work fine.

**Comment by Kevinrob** [comment]
Available in [v8.0.0-RC1](https://github.com/Kevinrob/guzzle-cache-middleware/releases/tag/v8.0.0-RC1)
